src/test/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerTimeEntryTest.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.homeoffice.digital.sas.timecard.producers;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.UUID;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.util.concurrent.ListenableFuture;
+import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
+import uk.gov.homeoffice.digital.sas.timecard.model.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
+
+@ExtendWith(SpringExtension.class)
+class KafkaProducerTimeEntryTest {
+
+  @Mock
+  private KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate;
+
+  @InjectMocks
+  private KafkaProducerTimeEntry kafkaProducerTimeEntry;
+
+  @ParameterizedTest
+  @EnumSource(value = KafkaAction.class, names = {"CREATE", "UPDATE"})
+  void sendMessage_actionOnTimeEntry_messageIsSentWithCorrectArguments(KafkaAction action) {
+    TimeEntry timeEntry = createTimeEntry();
+
+    ListenableFuture<SendResult<String, KafkaEventMessage>> responseFuture =
+        mock(ListenableFuture.class);
```

Comment:
This gives a `Unchecked assignment` warning. Can get rid of it by mocking `ListenableFuture.class` with an annotation at the instance variable level at the top.